debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisDatabaseHistory.java

```diff
@@ -0,0 +1,225 @@
+        this.config = config;
+        // fetch the properties. as a fallback, if we did not specify the redis address, we will take it and all the credentials from the sink
+        this.address = this.config.getString(PROP_ADDRESS.name());
+        if (this.address == null) {
+            // try to take the connection details from the redis sink
+            this.address = this.config.getString(SINK_PROP_PREFIX + "address");
+            this.user = this.config.getString(SINK_PROP_PREFIX + "user");
+            this.password = this.config.getString(SINK_PROP_PREFIX + "password");
+        }
+        else {
+            this.user = this.config.getString(PROP_USER.name());
+            this.password = this.config.getString(PROP_PASSWORD.name());
+        }
+        this.redisKeyName = this.config.getString(PROP_KEY);
+        LOGGER.info("rediskeyname:" + this.redisKeyName);
+        // load retry settings
+        this.initialRetryDelay = this.config.getInteger(PROP_RETRY_INITIAL_DELAY);
+        this.maxRetryDelay = this.config.getInteger(PROP_RETRY_MAX_DELAY);
+
+        super.configure(config, comparator, listener, useCatalogBeforeSchema);
+    }
+
+    @Override
+    public synchronized void start() {
+        super.start();
+        LOGGER.info("Starting RedisDatabaseHistory");
+        this.connect();
+    }
+
+    @Override
+    protected void storeRecord(HistoryRecord record) throws DatabaseHistoryException {
+        if (record == null) {
+            return;
+        }
+        try {
+            String line = writer.write(record.document());
+            Uni.createFrom().item(() -> {
+                return (StreamEntryID) client.xadd(this.redisKeyName, (StreamEntryID) null, Collections.singletonMap("schema", line));
+            })
+                    .onFailure().invoke(
+                            f -> {
+                                LOGGER.warn("Writing to database history stream failed with " + f);
+                                LOGGER.warn("Will retry");
+                            })
+                    .onFailure(JedisConnectionException.class).invoke(
+                            f -> {
+                                LOGGER.warn("Attempting to reconnect to redis ");
+                                this.connect();
+                            })
+                    // retry on failure with backoff
+                    .onFailure().retry().withBackOff(Duration.ofMillis(initialRetryDelay), Duration.ofMillis(maxRetryDelay)).indefinitely()
+                    // write success trace message
+                    .invoke(
+                            item -> {
+                                LOGGER.trace("Record written to database history in redis: " + line);
+                            })
+                    .await().indefinitely();
+        }
+        catch (IOException e) {
+            logger.error("Failed to convert record to string: {}", record, e);
```

Comment:
Let's use this class' own logger (i.e. `LOGGER`) instead, here and elsewhere.

Reply:
done